### PR TITLE
Less confusing stable channel recommendation

### PR DESCRIPTION
**Pull request by infinisil**
As mentioned in https://github.com/NixOS/nix.dev/pull/744#discussion_r1384292320

### CODE REVIEW

File: source/concepts/faq.md [hunk 1]
```diff
@@ -66,12 +66,13 @@ Nixpkgs and NixOS have both stable and rolling releases.
 
 ### Stable
 
-- On Linux (including NixOS and WSL), use [`nixos-*`](https://github.com/NixOS/nixpkgs/branches/all?query=nixos-).
+- On macOS/Darwin, use [nixpkgs-*-darwin](https://github.com/NixOS/nixpkgs/branches/all?query=nixpkgs-)
 
-  These branches only contain commits that passed the NixOS test suite.
-- On any other platform, use [`nixpkgs-*`](https://github.com/NixOS/nixpkgs/branches/all?query=nixpkgs-).
+  These branches point to commits where most Darwin packages can be substituted and don't have to be built.
+- On any other platform, use [`nixos-*`](https://github.com/NixOS/nixpkgs/branches/all?query=nixos-).
```
Comment (RaitoBezarius) [inline source/concepts/faq.md:72]: Not sure what is the value of using the NixOS channel by default, most people using Nixpkgs without NixOS won't care about the NixOS test suite.

How do we explain the rationale to someone else?

File: source/concepts/faq.md [hunk 2]
```diff
@@ -66,12 +66,13 @@ Nixpkgs and NixOS have both stable and rolling releases.
 
 ### Stable
 
-- On Linux (including NixOS and WSL), use [`nixos-*`](https://github.com/NixOS/nixpkgs/branches/all?query=nixos-).
+- On macOS/Darwin, use [nixpkgs-*-darwin](https://github.com/NixOS/nixpkgs/branches/all?query=nixpkgs-)
```
Comment (fricklerhandwerk) [inline source/concepts/faq.md:69]: Darwin is still a minority of our users, it doesn't really make sense to put this as the first item.

**Review by fricklerhandwerk — COMMENTED** [review]
I don't think this is an improvement. The review comment the structure of the current description is based on for a moment had, but did not end up with the bit, we discussed during review: that the current version is a meaningful simplification for beginners. It shouldn't matter at this point that we don't have anything but `darwin` branches in `nixpkgs-*`. We could add that fact though, so people aren't surprised.

File: source/concepts/faq.md (hunk 1 above)
Comment (infinisil) [reply source/concepts/faq.md:72]: We don't really have a choice right now. There's no stable Hydra channel that contains Linux builds but doesn't run the NixOS tests.

This is pretty much unchanged with this PR though, already before this section recommended the `nixos-*` channels to all Linux users, even if they're not on NixOS.

Comment (RaitoBezarius) [reply source/concepts/faq.md:72]: Why do you think https://hydra.nixos.org/job/nixpkgs/trunk/unstable#tabs-constituents does not contain Linux builds?

Comment (RaitoBezarius) [reply source/concepts/faq.md:72]: Ah a stable channel.

**Comment by infinisil** [comment]
Right now we're telling all non-Linux users to use the darwin branches, implicitly stating that these are the only two supported platforms. But really there's others like OpenBSD, which while Hydra doesn't build, are somewhat supported.

**Comment by infinisil** [comment]
How about doing this instead:
- On Linux, use `nixos-*`
- On Darwin, use `nixpkgs-*-darwin`
- On any other platform it doesn't matter which one of the above is used, Hydra doesn't pre-build any binaries for other platforms.

File: source/concepts/faq.md [hunk 3]
```diff
@@ -68,10 +68,16 @@ Nixpkgs and NixOS have both stable and rolling releases.
 
 - On Linux (including NixOS and WSL), use [`nixos-*`](https://github.com/NixOS/nixpkgs/branches/all?query=nixos-).
 
-  These branches only contain commits that passed the NixOS test suite.
-- On any other platform, use [`nixpkgs-*`](https://github.com/NixOS/nixpkgs/branches/all?query=nixpkgs-).
+  These branches point to commits where most Linux packages can be substituted and don't have to be built.
```
Comment (fricklerhandwerk) [inline source/concepts/faq.md:71]:
```suggestion
  These branches point to commits where most Linux packages got pre-built and can be fetched from the binary cache.
```

File: source/concepts/faq.md [hunk 4]
```diff
@@ -68,10 +68,16 @@ Nixpkgs and NixOS have both stable and rolling releases.
...
+  Furthermore, these commits passed the basic NixOS test suite.
 
-  These branches contain commits that passed the test suites specific to other platforms.
+- On macOS/Darwin, use [nixpkgs-*-darwin](https://github.com/NixOS/nixpkgs/branches/all?query=nixpkgs-)
+
+  These branches point to commits where most Darwin packages can be substituted and don't have to be built.
```
Comment (fricklerhandwerk) [inline source/concepts/faq.md:76]:
```suggestion
  These branches point to commits where most Darwin packages got pre-built and can be fetched from the binary cache.
```

No need to use fancy terms new users won't know.

**Review by fricklerhandwerk — APPROVED** [review]

**Pull request merged**